chingu-voyages/v49-tier3-team-29
# Feat/Connect password reset function with frontend

**Purpose:**
- Create UI to navigate user to forgot password page.
- Create UI to initiate password reset function through the URL that user received.

**Change(s):**
- Installed query-string to take token from password reset URL.
- Hard coded local server path in authController. (Reason: It needs the client's local path, not backend local path.)
- Added ForgotPasswordPage and ResetPasswordPage in App.jsx.
- Created new components for Forgot/Reset Password Pages.
- Added link to ForgotPasswordPage in LoginPage.jsx.

**How to review:**
![Video_2024-05-31_164658](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/ca6970f6-244f-4c23-816c-0c79ffe6953d)

1. Navigate to click "Forgot Password" link.
2. Input an existing user's email address that you can receive password reset email.
3. If email address exists in database, it sends email to reset password. If not, error "User not found" pops up.
4. Click the URL in the email.
5. Input a new password.
6. Try to log in with the new password you just set, and confirm the success response.

![Video_2024-05-31_171923](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/54f0cb44-fdf0-4d1c-9718-b782a035b7b5)
👆️ Error:  If token is invalid or expired, an error shows up. - _Invalid or expired password reset token_ 

![Video_2024-05-31_172127](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/97c7d556-a1c8-43e3-841a-429e176f5164)
👆️ Error: Email address does not match. - _User not found_
